manifests/init.pp

```diff
@@ -2,11 +2,79 @@
+# $admin_username::             Username for the initial admin user
+#
+# $admin_password::             Password of the initial admin user, default is randomly generated
+#
+# $admin_first_name::           First name of the initial admin user
+#
+# $admin_last_name::            Last name of the initial admin user
+#
+# $admin_email::                E-mail address of the initial admin user
+#
+# $db_manage::                  if enabled, will install and configure the database server on this host
+#                               type:boolean
+#
+# $db_type::                    Database 'production' type (valid types: mysql/postgresql/sqlite)
+#
+# $db_adapter::                 Database 'production' adapter
+#
+# $db_host::                    Database 'production' host
+#
+# $db_port::                    Database 'production' port
+#                               type:integer
+#
+# $db_database::                Database 'production' database (e.g. foreman)
+#
+# $db_username::                Database 'production' user (e.g. foreman)
+#
+# $db_password::                Database 'production' password (default is random)
+#
+# $db_sslmode::                 Database 'production' ssl mode
+#
+# $db_pool::                    Database 'production' size of connection pool
+#                               type:integer
+#
+# $email_conf::                 email configuration file, defaults to /etc/foreman/email.yaml
+#
+# $email_source::               template to use for email configuration file
+#
+# $email_delivery_method::      can be sendmail or smtp regarding to foreman documentation
+#
+# $email_smtp_address::         if delivery_method is smtp, this should contain an valid smtp host
+#
+# $email_smtp_port::            smtp port, defaults to 25
+#                               type: integer
+#
+# $email_smtp_domain::          email domain
+#
+# $email_smtp_authentication::  authentication settings, can be none or login, defaults to none
+#
+# $email_smtp_user_name::       user_name for mail server auth, if authentication login
+#
+# $email_smtp_password::        password for mail server auth, if authentication login
+#
+#
+# $locations_enabled::          Enable locations
+#                               type:boolean
+#
+# $organizations_enabled::      Enable organizations
+#                               type:boolean
+#
+# $initial_organization::       Name of an initial organization
+#
+# $initial_location::           Name of an initial location
+#
+# $ipa_authentication::         Enable configuration for external authentication via IPA
+#                               type:boolean
 #
 # $puppetrun::                  Should foreman be able to start puppetruns on nodes
 #                               type: boolean
 #
+# === Advanced parameters:
+#
+# $foreman_url::                URL on which foreman is going to run
+#
 # $unattended::                 Should foreman manage host provisioning as well
```

Comment:
Should it? I don't mind moving it, but in my mind, I put this is the same category as 'authorization' - these legacy components that you *can* turn off in Settings, but plugins or even core configuration may misbehave if you do it because we don't particularly test well in those situations.